### ISSUE TITLE
How do I set my Drupal site URI in Lando for easy Drushing?

### ISSUE THREAD

**Issue by dustinleblanc** [lando/lando#2080]
As a follow up to #447, we should write up a guide on how to modify Drush options within Lando to make `uli` use the correct domain so that users can make their Lando setup just that much nicer.

**Comment by gitressa** [comment]
There is already this page, or do you think it should be expanded?
https://docs.lando.dev/config/drupal8.html#configuration

**Comment by serundeputy** [comment]
@gitressa yep that doc has it and also a significant amount of other info. 

The goal here is to pull that out (leave it where it is too) as a very specific `guide` to answer the one question `How do I set my Drupal site URI in Lando for easy Drushing?`, but maybe we could work on the wording of that question, but that is the question.

**Comment by gitressa** [comment]
Thanks for clarifying @serundeputy. I just discovered another and slightly shorter way of setting the Drupal site URI:

```
tooling:
  drush:
    service: appserver
    cmd: drush --root=/app/web --uri=http://mysite.lndo.site
```
Compared to this:
```
services:
  appserver:
    overrides:
      environment:
        DRUSH_OPTIONS_URI: "http://mysite.lndo.site"
```

I wonder if there are reasons to use one over the other?

**Comment by gitressa** [comment]
To answer my own question, it appears that defining Drush URI via `tooling` will not work in another instance of tooling:
```
tooling:
  drush:
    service: appserver
    cmd: drush --root=/app/web --uri=http://mysite.lndo.site
  post-import:
    service: appserver
    cmd: drush uli
```
The result is http://default/user/reset/1/1586688025/TOp-0F....XxQ/login

**Comment by dustinleblanc** [comment]
I'd prefer the env variable approach, it seems cleaner and more intentional

**Comment by gitressa** [comment]
Yes, and it's the one that will actually work inside another `tooling` instance, whereas defining Drush URI in `tooling` doesn't. So unless that gets fixed, it's not very useful.

**Comment by AayamBhatt** [comment]
it can be useful if modified in creative way .

**Comment by mdrmike** [comment]
> There is already this page, or do you think it should be expanded?
> https://docs.lando.dev/config/drupal8.html#configuration

From Lando homepage, the point is to **"Save time, headaches, frustration and do more real work."** 

In that spirit I feel there should be complete real-world workable examples for **Drush 8** and **Drush 9+**; the latter use `yaml alias files` not `php alias` files. Each should also have examples of **global-site** and **local-site** setup (in addition to @self setup, including examples of a remote server setup would be a bonus for some newer folks; e.g. @staging or @production). 

So yes, I feel it should be expanded/clarified. I think some of the documentation is pretty spectacular. But Drush aliases has been particularly confusing to me, someone with previous Drupal/Drush experience, but new to Drupal 8 and trying to figure out what Lando is doing behind the scenes. I'm working to deploy my first Drupal 8.8 site using composer, with Drush 9. My experience setting up Drush aliases has been, erm, $#%@ difficult. :grimacing: 

It took me (multiple!) tries/days reading the Lando Drush docs before I understood what is meant by a global install -vs- local install. The first couple reads of the Lando docs suggested to me that global install meant an existing Drupal / Drush setup environment that is outside/agnostic of Lando ... as both recipe and composer install of Drush seem to be "local" to the Lando app. It also didn't help that external examples to [install a site-local Drush](https://docs.drush.org/en/master/install/) defaults to Drush 10 examples (master branch), which are _totally different than anything else_ on the Lando documents page. 

The difficult thing for me has been trying to discover/debug the $PATH that Drush uses in order to figure out where to put the file, esp when the recipe also installs a version of Drush in addition to a composer version. Its a tangled web.

**Side note:** It also seems to me that Lando (and documentation in general) has yet to embrace a composer based workflow. Although a separate issue, I feel this may be part of the problem ... Drush 9 is composer based, release 2.5yrs ago. Lando should more easily/clearly support newer workflows.

Anyway, that's my two cents. Hope I didn't come across as rude. Don't hate me... just trying to share what I feel could be improved. :smiling_face_with_three_hearts:

**Comment by pirog** [comment]
@mdrmike ill be honest i dont really understand what you are saying in most of the above comment. 

@dustinleblanc @serundeputy there are two new things to consider here

1. You can now specify envvars for tooling commands eg

```yaml
tooling:
  drush:
    service: appserver
    cmd: drush
    env:
      DRUSH_OPTIONS_URI: mysite.lndo.site
```

**Comment by mdrmike** [comment]
@pirog Ha! Sorry. I was confused with the issue and some terms when I commented. 

I was piling on with detailed explanation. My comment was an attempt to show where I had problems with the config docs and make the case for why its important to have beginning-to-end examples, both a basic and complex example. But I think that's the intention of guides. Right?

Also, the words `guide`, `docs` (in url: https://docs.lando.dev/config/drupal8.html), and `config` didn't have enough distinction for me to realize I was seeing/reading multiple things. It took multiple re-reads. Sorry, its just the lando lingo that I'm sure is second nature to most of you... but I finally figured out they had specific meanings.

**Comment by markdorison** [comment]
Is there a solution for this that would work in a multi-site setup? Maybe that is the proxy option? It's not 100% clear to me how that connects.

**Comment by dustinleblanc** [comment]
@markdorison when dealing with Drupal multisite, I don't see a way around passing -l with every drush command unless you add multiple tooling commands that string it on to drush itself. Its been ages since I had to work on a multi-site, but I remember having to do that to run things

**Comment by gitressa** [comment]
Thanks Stalebot, but this issue is still relevant.

**Comment by froboy** [comment]
One other consideration here is ports. Drush needs the correct port to be passed to DRUSH_OPTIONS_URI and unless it’s hard-coded in .lando.yml the port could be different depending on what lando grabs when it starts up the site.

I’m currently struggling with this trying to run lando and docksal in parallel (not forever, hopefully…).
